src/pages/IstioConfigList/FiltersAndSorts.ts

```diff
@@ -0,0 +1,169 @@
+      },
+      {
+        id: 'VirtualService',
+        title: 'VirtualService'
+      },
+      {
+        id: 'DestinationRule',
+        title: 'DestinationRule'
+      },
+      {
+        id: 'ServiceEntry',
+        title: 'ServiceEntry'
+      },
+      {
+        id: 'Rule',
+        title: 'Rule'
+      },
+      {
+        id: 'QuotaSpec',
+        title: 'QuotaSpec'
+      },
+      {
+        id: 'QuotaSpecBinding',
+        title: 'QuotaSpecBinding'
+      }
+    ]
+  };
+
+  const configValidationFilter: FilterType = {
+    id: 'configvalidation',
+    title: 'Config',
+    placeholder: 'Filter by Config Validation',
+    filterType: 'select',
+    action: FILTER_ACTION_APPEND,
+    filterValues: [
+      {
+        id: 'valid',
+        title: 'Valid'
+      },
+      {
+        id: 'warning',
+        title: 'Warning'
+      },
+      {
+        id: 'notvalid',
+        title: 'Not Valid'
+      },
+      {
+        id: 'notvalidated',
+        title: 'Not Validated'
+      }
+    ]
+  };
+
+  export const availableFilters: FilterType[] = [
+    NamespaceFilter.create(),
+    istioTypeFilter,
+    istioNameFilter,
+    configValidationFilter
+  ];
+
+  export const sortIstioItems = (
+    unsorted: IstioConfigItem[],
+    sortField: SortField<IstioConfigItem>,
+    isAscending: boolean
+  ) => {
+    const sortPromise: Promise<IstioConfigItem[]> = new Promise((resolve, reject) => {
+      resolve(
+        unsorted.sort((a: IstioConfigItem, b: IstioConfigItem) => {
+          let sortValue = -1;
+          if (sortField.id === 'namespace') {
+            sortValue = a.namespace.localeCompare(b.namespace);
+          }
+          if (sortField.id === 'istiotype') {
+            sortValue = a.type.localeCompare(b.type);
+          }
+          if (sortField.id === 'configvalidation') {
+            if (a.validation && !b.validation) {
+              sortValue = -1;
+            }
+            if (!a.validation && b.validation) {
+              sortValue = 1;
+            }
+            if (!a.validation && !b.validation) {
+              sortValue = 0;
+            }
+            if (a.validation && b.validation) {
+              if (a.validation.valid && !b.validation.valid) {
+                sortValue = -1;
+              }
+              if (!a.validation.valid && b.validation.valid) {
+                sortValue = 1;
+              }
+              if (a.validation.valid && b.validation.valid) {
+                sortValue = 0;
+              }
+              if (!a.validation.valid && !b.validation.valid) {
+                sortValue = b.validation.checks.length - a.validation.checks.length;
+              }
+            }
+          }
+          // Istioname at the end to be the default sort when sortValue === 0
+          if (sortField.id === 'istioname' || sortValue === 0) {
+            sortValue = a.name.localeCompare(b.name);
+          }
+          return isAscending ? sortValue : sortValue * -1;
+        })
```

Comment:
you are right, we only need to run one sort at a time :) Good catch!
On the other side, to reduce complexity and increase OOness, SortFields has compare func which handles the sort algorithm. So, the refactor could be as it follows:

```typescript
    return new Promise((resolve, reject) => {
      resolve(
        unsorted.sort(isAscending ? sortField.compare : (a, b) => sortField.compare(b, a))
      );
    });
```

However, sorting by config validation doesn't work after this refactoring because it needs the validations (fetched in parallel) beforehand. It looks like sorting objects as it is now in the PR takes enough time for the sorting logic to have validations in place and perform that sorting.

I think I would leave this code as it is now, since is the original written code that couldn't be unified. what do you think?

cc @lucasponce @aljesusg

Reply:
Hmm, I'm not sure to understand exactly: you mean that because of some higher processing time it happens to work in the right order?
That would be a concern. The promise stuff is precisely here to allow asynchronous coordination on sort.

But ok, on the other hand since it's a quite big refactoring it could be hard to maintain & rebase if we wait too much to merge it, so I'm ok if we say that we keep this potential problem under our radar and fix it afterward.